server/game/cards/03_TWI/units/EnfysNestChampionOfJustice.ts

```diff
@@ -0,0 +1,31 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { RelativePlayer, WildcardCardType } from '../../../core/Constants';
+
+export default class EnfysNestChampionOfJustice extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '8414572243',
+            internalName: 'enfys-nest#champion-of-justice',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addTriggeredAbility({
+            title: 'Return an enemy non-leader unit with less power than this unit to its owner\'s hand',
+            when: {
+                onCardPlayed: (event, context) => event.card === context.source,
+                onAttackDeclared: (event, context) => event.attack.attacker === context.source,
+            },
+            optional: true,
+            targetResolver: {
+                controller: RelativePlayer.Opponent,
+                cardTypeFilter: WildcardCardType.NonLeaderUnit,
+                cardCondition: (card, context) => card.isUnit() && card.getPower() < context.source.getPower(),
```

Comment:
If we have to have a `card.is*()` check anyway we may as well remove the NonLeaderUnit filter and just check `card.isNonLeaderUnit()` here